py/picca/delta_extraction/data_catalogues/desi_healpix.py

```diff
@@ -125,58 +129,69 @@ def read_data(self):
 
         is_sv = True
         is_mock = False
+        forests_by_targetid = {}
+
+        arguments = []
+        for (healpix, survey), group in zip(grouped_catalogue.groups.keys,
+                                            grouped_catalogue.groups):
+            if survey not in ["sv", "sv1", "sv2", "sv3"]:
+                is_sv = False
+
+            filename, is_mock_aux = self.get_filename(survey, healpix)
+            if is_mock_aux:
+                is_mock = True
+
+            arguments.append((filename, group))
+
+        self.logger.info(f"reading data from {len(arguments)} files")
+
         if self.num_processors > 1:
             context = multiprocessing.get_context('fork')
-            manager = multiprocessing.Manager()
-            forests_by_targetid = manager.dict()
-            arguments = []
-            for (index, (healpix, survey)), group in zip(
-                    enumerate(grouped_catalogue.groups.keys),
-                    grouped_catalogue.groups):
-
-                if survey not in ["sv", "sv1", "sv2", "sv3"]:
-                    is_sv = False
-
-                filename, is_mock_aux = self.get_filename(survey, healpix)
-                if is_mock_aux:
-                    is_mock = True
-                arguments.append((filename, group, forests_by_targetid))
-
-                self.logger.info(f"reading data from {len(arguments)} files")
             with context.Pool(processes=self.num_processors) as pool:
+                imap_it = pool.imap(
```

Comment:
I also recently changed this to imap_unordered. Shouldn't be a big deal either way

Reply:
How do imap and imap_unordered differ? Is it a matter of speed? If it's not significant I'd rather keep the ordered version to ensure that it runs the same way in different machines. This would make things simpler to debug. For example, it took me a while to figure out that the delta_extraction code was not giving me the exact same results because the spectra were ordered in a different way and the DLA masking had some random numbers involved, meaning that the order mattered

Reply:
I think the difference is imap preserves the order of the arguments, and imap_unordered doesn't, which might speed up a bit. But yours is a valid reason to keep it ordered. There might be subtleties when data gets bigger, but no need to prematurely worry about it.